Tests/Unit/HttplugFactoryTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Http\HttplugBundle\Tests\Unit\ClientFactory;
+
+use Http\HttplugBundle\Collector\MessageJournal;
+use Http\HttplugBundle\HttplugFactory;
+use Puli\Discovery\Api\Discovery;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttplugFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testEvaluateCondition()
+    {
+        $method = new \ReflectionMethod(HttplugFactory::class, 'evaluateCondition');
```

Comment:
> Your suggestion would require a lot of mocking and is very likely to break if the class changes.

What class?

I think it requires an acceptable amount of mocks. Not to mention you can reuse mock builders. (Even better: as of PHPUnit 4.5 you can use prophecy known from phpspec. You could even use phpspec in httplugbundle as well along with phpunit, that would probably be even better)